ext/standard/password.c

```diff
@@ -61,7 +81,14 @@ static php_password_algo php_password_determine_algo(const char *hash, const siz
 {
 	if (len > 3 && hash[0] == '$' && hash[1] == '2' && hash[2] == 'y' && len == 60) {
 		return PHP_PASSWORD_BCRYPT;
+	} 
+#if HAVE_ARGON2LIB
+	if (hash[0] == '$' && strstr(hash, "argon2i")) {
```

Comment:
I don't think it matters since bcrypt and crypt hashes are checked before hand, but I went ahead and converted it to your suggestion anyways. Your solution makes this a little more portable if new algorithms are added.